docs/MachineConfigDaemon.md

```diff
@@ -170,4 +170,46 @@ The updated list of optimized updates and behaviour (as of Openshift 4.8) is as
 
 ## Annotating on SSH access
 
-RHCOS nodes in Openshift are not meant to be manually accessed via SSH. MCD uses logind to watch for login sessions, which, upon detection, warns the user and annotates the node with `machineconfiguration.openshift.io/ssh=accessed`. This in turn will be used to warn cluster admins.
\ No newline at end of file
+RHCOS nodes in Openshift are not meant to be manually accessed via SSH. MCD uses logind to watch for login sessions, which, upon detection, warns the user and annotates the node with `machineconfiguration.openshift.io/ssh=accessed`. This in turn will be used to warn cluster admins.
+
+## Config Drift Detection
+
+### Overview
+
+There may be situations when the on-disk state of a node may differ or "drift"
+from what is specified by the MachineConfig. This is usually (though not
+always) due to a cluster admin manually editing files. In the past, the MCD
+would only verify that the on-disk state matches what is specified by the
+MachineConfig after a reboot. Whenever this occurs, the node and Machine Config
+Pool (MCP) would be marked Degraded and unable to apply any MachineConfigs.
+
+Because reboots do not frequently occur, the config drift isn't immediately
+detected. In cases where the config drift was intentional, the cluster admin
+may lose the context which necessitated the config drift between the time the
+drift occurred and when it was detected. This leads to the node and / or MCP
+becoming degraded at an inconvenient time and can block future updates until
+the issue is remedied.
+
+### Detection
+
+To avoid this, as of 4.10, the MCD uses fsnotify to proactively detect config
+drift and mark the node Degraded within seconds of the drift taking place. It
+does so with the following rules:
+
+- It only pays attention to files, systemd units / dropins belonging to the
+currently applied MachineConfig. In particular, it pays attention to the
+contents of the file, the permissions of the file, and whether it exists. 
+- If a file is not defined by the currently applied MachineConfig, that file
+will be ignored by the config drift detection.
+
+### Remediation
+
+Once config drift is detected, there are two ways to recover:
```

Comment:
I'm curious when each action will be taken

Reply:
I reworded most of this so that it better reflects what the Config Drift Monitor looks for, how it detects it, and what actions it takes. I also added additional context around how updates occur with the presence of the Config Drift Monitor.

Reply:
Might be too technical to share these details. But I'm still curious:
1. Is this section step 5 for above? Or when does remediation occur?
2. In this section, when are the contents just rewritten, and when is the forcefile created?

Your call on whether that's worth adding!